includes/discovery/sensors/state/cisco.inc.php

```diff
@@ -135,26 +136,31 @@
         }
 
         foreach ($temp as $index => $entry) {
-            //Discover Sensors
-            $descr = ucwords($temp[$index][$tablevalue[3]]);
-            if ($state_name == 'cRFStatusUnitState' || $state_name == 'cRFStatusPeerUnitState' || $state_name == 'cRFCfgRedundancyOperMode' || $state_name == 'cswRingRedundant') {
-                $descr = $tablevalue[3];
-            } elseif ($state_name == 'cswSwitchRole') {
-                $swrolenumber++;
-                $descr = $tablevalue[3] . $swrolenumber;
-            } elseif ($state_name == 'cswSwitchState') {
-                $swstatenumber++;
-                $descr = $tablevalue[3] . $swstatenumber;
-            } elseif ($state_name == 'cswStackPortOperStatus') {
-                $stack_port_descr = get_port_by_index_cache($device, $index);
-                $descr = $tablevalue[3] . $stack_port_descr['ifDescr'];
-            } elseif ($state_name == 'cefcFRUPowerOperStatus') {
-                $descr = snmp_get($device, 'entPhysicalName.'.$index, '-Oqv', 'ENTITY-MIB');
+            if ($tablevalue[2] == 'ciscoEnvMonTemperatureState' && (is_null($temp[$index][$tablevalue[3]]) or empty($temp[$index][$tablevalue[3]]))) {
+                $process_sensor = false;
             }
-            discover_sensor($valid['sensor'], 'state', $device, $cur_oid.$index, $index, $state_name, $descr, '1', '1', null, null, null, null, $temp[$index][$tablevalue[2]], 'snmp', $index);
+            if ($process_sensor == true) {
```

Comment:
Why not just leave this to the previous `if()` check that I commented on, seems pointless setting a variable.

Reply:
Not sure what you mean here.
What are you suggesting?

Reply:
Litterally just change it from:

```php
}
if ($process_sensor == true) {
```

To:

```php
} else {
```

No point in assigning a variable in one if statement only to open a new one using that variable.

Reply:
Now I understand what you mean, you are totally right.